coordinator/provider/coordinator.go

```diff
@@ -256,22 +256,39 @@ func (qc *qdbCoordinator) watchRouters(ctx context.Context) {
 	}
 }
 
-// NewCoordinator side efferc: runs async goroutine that checks
-// spqr router`s availability
 func NewCoordinator(db qdb.XQDB) *qdbCoordinator {
-	cc := &qdbCoordinator{
+	return &qdbCoordinator{
 		db: db,
 	}
+}
+
+// RunCoordinator side efferc: runs async goroutine that checks
+// spqr router`s availability
+func (cc *qdbCoordinator) RunCoordinator(ctx context.Context) {
+	if cc.db.TryCoordinatorLock(context.TODO()) != nil {
+	LOCK:
```

Comment:
the wat
Lets rewrite this method without using LOCK tag
Or just move TryCoordinatorLock to a separate method